crates/mun_hir/src/name_resolution/path_resolution.rs

```diff
@@ -0,0 +1,149 @@
+            segment_index,
+            package,
+        }
+    }
+}
+
+impl PackageDefs {
+    /// Resolves the specified `path` from within the specified `module`. Also optionally returns
+    /// which part of the path was resolved, if this is not `None` it means the path didn't resolve
+    /// completely yet.
+    pub(crate) fn resolve_path_in_module(
+        &self,
+        db: &dyn DefDatabase,
+        module: LocalModuleId,
+        path: &Path,
+    ) -> (PerNs<(ItemDefinitionId, Visibility)>, Option<usize>) {
+        let res = self.resolve_path_with_fixedpoint(db, module, path);
+        (res.resolved_def, res.segment_index)
+    }
+
+    /// Resolves the specified `name` from within the specified `module`
+    fn resolve_name_in_module(
+        &self,
+        _db: &dyn DefDatabase,
+        module: LocalModuleId,
+        name: &Name,
+    ) -> PerNs<(ItemDefinitionId, Visibility)> {
+        self[module]
+            .get(name)
+            .or(BUILTIN_SCOPE.get(name).copied().unwrap_or_else(PerNs::none))
+    }
+
+    /// Resolves the specified `path` from within the specified `module`. Also returns whether or
+    /// not an additions to the PackageDef would change the result. This is useful when resolving
+    /// all imports.
+    pub(super) fn resolve_path_with_fixedpoint(
+        &self,
+        db: &dyn DefDatabase,
+        original_module: LocalModuleId,
+        path: &Path,
+    ) -> ResolvePathResult {
+        let mut segments = path.segments.iter().enumerate();
+        let mut curr_per_ns: PerNs<(ItemDefinitionId, Visibility)> = match path.kind {
+            PathKind::Plain => {
+                let (_, segment) = match segments.next() {
+                    Some((idx, segment)) => (idx, segment),
+                    None => return ResolvePathResult::empty(ReachedFixedPoint::Yes),
+                };
+                self.resolve_name_in_module(db, original_module, &segment)
+            }
+            PathKind::Super(lvl) => {
+                let m = successors(Some(original_module), |m| self.module_tree[*m].parent)
+                    .nth(lvl as usize);
+                if let Some(local_id) = m {
+                    PerNs::types((
+                        ModuleId {
+                            package: self.module_tree.package,
+                            local_id,
+                        }
+                        .into(),
+                        Visibility::Public,
+                    ))
+                } else {
+                    return ResolvePathResult::empty(ReachedFixedPoint::Yes);
+                }
+            }
+            PathKind::Package => PerNs::types((
+                ModuleId {
+                    package: self.module_tree.package,
+                    local_id: self.module_tree.root,
+                }
+                .into(),
+                Visibility::Public,
+            )),
+            PathKind::Abs => {
+                // TODO: How do we get here?
```

Comment:
If so, can this be an `unreachable!`?

Reply:
I removed the `PathKind::Abs` altogether. I don't see a use-case for it.